src/7factor.io/converter/ecs.go

```diff
@@ -14,16 +13,118 @@ type Pair struct {
 	Value string `json:"value"`
 }
 
+
 func TransformAndTranslate(contents string) (string, error) {
 	if contents == "" {
 		return `[]`, errors.New("contents cannot be empty")
 	}
 
-	cleanedContents := cleanContents(contents)
-	pairs := transform(cleanedContents)
+	var itemsToBeParsed []string
+	for _, line := range strings.Split(contents, "\n") {
+		if hasComment(line) || hasEmptyString(line) {
+			continue
+		}
+		items := parse(line)
+		
+
+		itemsToBeParsed = append(itemsToBeParsed, items...)
+	}
+
+	pairs := transform(itemsToBeParsed)
 	return translate(pairs)
 }
 
+var commentRegex = regexp.MustCompile("(?m)^\\s*\\#.*$")
+func hasComment(line string) bool {
+	return commentRegex.MatchString(line)
+}
+
+func hasEmptyString(line string) bool {
+	return len(line) == 0 || line == ""
+}
+
+var singleValueContextRegex = regexp.MustCompile("^.+?=.+?$")
+func parse(line string) []string {
```

Comment:
Wow this is a crazy function. Even though it's long, since it's async this might be the best way to write it. I might suggest a more descriptive name than `parse`. Perhaps `runHandlers`. This is almost the true `Transform` function, so maybe `TransformAndTranslate needs to be refactored.